libs/s25main/ogl/glSmartBitmap.cpp

```diff
@@ -240,12 +242,32 @@ void glSmartBitmap::generateTexture()
     }
 }
 
+void glSmartBitmap::Draw(Rect dstArea, Rect srcArea, unsigned color /*= 0xFFFFFFFF*/)
+{
+    drawRect(dstArea, srcArea, color);
+}
+
 void glSmartBitmap::draw(DrawPoint drawPt, unsigned color, unsigned player_color)
 {
     drawPercent(drawPt, 100, color, player_color);
 }
 
 void glSmartBitmap::drawPercent(DrawPoint drawPt, unsigned percent, unsigned color, unsigned player_color)
+{
+    // nothing to draw?
+    if(!percent)
+        return;
+    RTTR_Assert(percent <= 100);
```

Comment:
I think this should be tested too. Especially coming up with a few corner cases. I.e. at least percent = 0 or 100 and some values where you notice the (correct) usage of floor/ceil below. I.e. choose appropriate image size and percent values.

Reply:
I've added some unit test here, but I'm not quite sure on how I should approach the float/ceil thingy - I can't really subclass `glSmartBitmap` and mock `drawRect` unless I make it virtual, which I think shouldn't be done just for unit tests. The texture coords don't follow image size in pixels, for example I tried to create 50x100 bitmap and coord span for the texture was something like 0 .. 0.7825. Checking for magic float values seems not to be the best approach either.